File: next-app/src/lib/deviceId.js
// Stable per-browser device id for the 2-device login cap. A random UUID kept in
// localStorage; clearing site storage rotates it (which then counts as a new
// device — recover with an admin "reset devices"). Client-only.

export function getOrCreateWebDeviceId() {
  if (typeof window === 'undefined') return null;
  try {
    let id = localStorage.getItem('deviceId');
    if (!id) {
      id =
        (typeof crypto !== 'undefined' && crypto.randomUUID)
          ? crypto.randomUUID()
          : `web-${Date.now()}-${Math.random().toString(36).slice(2)}`;
      localStorage.setItem('deviceId', id);
    }
    return id;
  } catch {
    return null;
  }
}

// A short human label for the device list in the admin UI.
export function webDeviceName() {
  if (typeof navigator === 'undefined') return null;
  const ua = navigator.userAgent || '';
  return ua ? ua.slice(0, 160) : null;
}

File: android/app/src/main/java/com/attenddesk/data/DeviceId.kt
package com.attenddesk.data

import android.content.Context
import java.util.UUID

/**
 * Stable per-install device id for the server's 2-device login cap. A random UUID
 * kept in SharedPreferences (read synchronously so the OkHttp interceptor can add
 * it without blocking on DataStore). Reinstalling/clearing app data rotates it —
 * which then counts as a new device; recover with an admin "reset devices".
 */
object DeviceId {
    private const val PREFS = "device_prefs"
    private const val KEY = "device_id"

    fun get(context: Context): String {
        val sp = context.getSharedPreferences(PREFS, Context.MODE_PRIVATE)
        var id = sp.getString(KEY, null)
        if (id.isNullOrBlank()) {
            id = UUID.randomUUID().toString()
            sp.edit().putString(KEY, id).apply()
        }
        return id
    }
}
